### PR TITLE
Improve log level conventions

### PR DESCRIPTION
**Pull request by jodavis-suse**
Referencing the upstream Kubernetes Instrumentation
Logging Conventions, move some of the log messages
in Skuba to align with described levels.

[0] https://github.com/kubernetes/community/blob/master/contributors
/devel/sig-instrumentation/logging.md

## Why is this PR needed?

https://github.com/SUSE/avant-garde/issues/1132

## What does this PR do?

The upstream Kubernetes Instrumentation SIG has published a convention for setting logging levels.  This PR attempts to get skuba code in line with those conventions (with values in the 0-5 range) by reviewing the existing klog.V() calls (which currently are all klog.V(1)) and setting levels for the messages that align with the convention.

## Anything else a reviewer needs to know?

There should be no significant change to user experience, though some messages that are currently logged with a 1 will now require a higher log level to trigger.

## Info for QA

Running skuba with -v [2-5] may not produce more log output than just -v 1, depending on the command and code path invoked.

### Status **BEFORE** applying the patch

Effectlively, all Info messages from skuba internal code were logged with a 1 level.

### Status **AFTER** applying the patch

Running skuba with -v [2-5] may not produce more log output than just -v 1, depending on the command and code path invoked.

## Docs

No documentation impact due to this change - current documentation about log levels is still correct.

# Merge restrictions

(Please do not edit this)

We are in *v4-maintenance phase*, so we will restrict what can be merged to prevent unexpected surprises:

    What can be merged (merge criteria):
        2 approvals:
            1 developer: code is fine
            1 QA: QA is fine
        there is a PR for updating documentation (or a statement that this is not needed)

### CODE REVIEW

File: internal/pkg/skuba/deployments/ssh/ssh.go [hunk 1]
```diff
@@ -165,7 +165,7 @@ func (t *Target) internalSshWithStdin(silent bool, stdin string, command string,
 		finalCommand = fmt.Sprintf("sudo sh -c '%s'", finalCommand)
 	}
 	if !silent {
-		klog.V(1).Infof("running command: %q", finalCommand)
+		klog.V(2).Infof("running command: %q", finalCommand)
```
Comment (jodavis-suse) [inline internal/pkg/skuba/deployments/ssh/ssh.go]: Actually, these log messages in ssh.go make me nervous, as I don't see any sort of security check in place that would prevent any sensitive information used in the ssh command from being logged.  I am hoping others with more experience in skuba can comment.

File: internal/pkg/skuba/deployments/ssh/ssh.go [hunk 2]
```diff
@@ -188,7 +188,7 @@ func readerStreamer(reader io.Reader, outputChan chan<- string, description stri
 	for scanner.Scan() {
 		result.Write([]byte(scanner.Text()))
 		if description == "stdout" && !silent {
-			klog.V(1).Infof("%s", scanner.Text())
+			klog.V(2).Infof("%s", scanner.Text())
```
Comment (jodavis-suse) [inline internal/pkg/skuba/deployments/ssh/ssh.go]: Again, slight security paranoia flag here as we are logging all the output from the ssh command.

File: internal/pkg/skuba/kured/kured.go [hunk 3]
```diff
@@ -44,7 +44,7 @@ func Lock(client clientset.Interface) error {
 	if err != nil {
 		return errors.Wrap(err, "unable to patch daemonset with kured locking annotation")
 	}
-	klog.V(1).Info("successfully annotated daemonset with kured locking annotation")
+	klog.V(2).Info("successfully annotated daemonset with kured locking annotation")
```
Comment (jodavis-suse) [inline internal/pkg/skuba/kured/kured.go]: I wasn't clear on the frequency or importance of the log messages in kured.go.  If they are rarely logged and of high importance, then leaving them at V(1) may be appropriate.

File: internal/pkg/skuba/deployments/ssh/ssh.go (hunk 2 above)
Comment (suseclee) [reply internal/pkg/skuba/deployments/ssh/ssh.go]: I believe, most of logs comes from this line. Good to have differentiate log level.
I was confusing v(2) and V(3). I read doc again. klog.V(2) - Useful steady state information about the service and important log messages that may correlate to significant changes in the system. I think this satisfies V(2).

**Comment by jodavis-suse** [comment]
Rebased to resolve a merge conflict.  Hope that resolves any check failures too.

File: internal/pkg/skuba/deployments/ssh/ssh.go (hunk 1 above)
Comment (Klaven) [reply internal/pkg/skuba/deployments/ssh/ssh.go]: there is a "silent" ssh. this is the one we use for secure stuff that does not log anything.

File: internal/pkg/skuba/kured/kured.go (hunk 3 above)
Comment (Klaven) [reply internal/pkg/skuba/kured/kured.go]: IMO 2 is fine here.

File: internal/pkg/skuba/deployments/ssh/ssh.go (hunk 1 above)
Comment (jodavis-suse) [reply internal/pkg/skuba/deployments/ssh/ssh.go]: Good point, and good to know where we place the trust that the command is not run incorrectly.

**Pull request merged**